### PR TITLE
Remove unused devbot

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Removes triage devbot and docs link workflow.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [RFC] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: .github/workflows/docs-link.yml [hunk 1]
```diff
@@ -1,14 +0,0 @@
-name: Docs Link
```
Comment (rohitgr7) [inline .github/workflows/docs-link.yml:1]: do you know why this isn't displayed in the ci section?

Comment (akihironitta) [reply .github/workflows/docs-link.yml:1]: Let me have a look again...

**Comment by akihironitta** [comment]
@carmocca Just an idea, but since devbot didn't work, could we maybe have a default label "needs triage" for all issue templates? In this way, core contributors can remove the default label and add other labels when creating issues.

**Review by justusschock — APPROVED** [review]

**Comment by carmocca** [comment]
@akihironitta we can do this as a simpler solution for the time being since the bot did not work.

**Review by Borda — APPROVED** [review]

File: .github/workflows/docs-link.yml (hunk 1 above)
Comment (akihironitta) [reply .github/workflows/docs-link.yml:1]: Reverting the change in this file as I might've found a cause... Let me give it a try in #12449.

**Review by krshrimali — APPROVED** [review]
🚀

**Pull request merged**